test/extended/util/nfs.go

```diff
@@ -0,0 +1,170 @@
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"name": name},
+		},
+		Spec: kapiv1.ReplicationControllerSpec{
+			Replicas: &replicas,
+			Selector: map[string]string{
+				"role": "nfs-server",
+			},
+			Template: &kapiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"role": "nfs-server"},
+				},
+				Spec: kapiv1.PodSpec{
+					Containers: []kapiv1.Container{
+						{
+							Name:  "nfs-server",
+							Image: "docker.io/coreydaley/nfs-server",
+							Ports: []kapiv1.ContainerPort{
+								{ContainerPort: 2049, Name: "nfs"},
+								{ContainerPort: 2049, Name: "nfs-udp", Protocol: "UDP"},
+								{ContainerPort: 20048, Name: "mountd"},
+								{ContainerPort: 111, Name: "rpcbind"},
+								{ContainerPort: 111, Name: "rpcbind-udp", Protocol: "UDP"},
+							},
+							Resources: kapiv1.ResourceRequirements{
+								Requests: kapiv1.ResourceList{
+									kapiv1.ResourceMemory: resource.MustParse(capacity),
+								},
+							},
+							SecurityContext: &kapiv1.SecurityContext{
+								Privileged: &privileged,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// CreateNFSServerService creates a service for the nfs replication controller
+func CreateNFSServerService(name string) *kapiv1.Service {
+	return &kapiv1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"name": name},
+		},
+		Spec: kapiv1.ServiceSpec{
+			Ports: []kapiv1.ServicePort{
+				{Port: 2049, Name: "nfs"},
+				{Port: 2049, Name: "nfs-udp", Protocol: "UDP"},
+				{Port: 20048, Name: "mountd"},
+				{Port: 111, Name: "rpcbind"},
+				{Port: 111, Name: "rpcbind-udp", Protocol: "UDP"},
+			},
+			Selector: map[string]string{
+				"role": "nfs-server",
+			},
+		},
+	}
+}
+
+// SetupNFSServer sets up an nfs server replication controller with the given capacity
+// and the nfs service
+func SetupNFSServer(oc *CLI, capacity string) (*kapiv1.ReplicationController, *kapiv1.Service, error) {
+	e2e.Logf("Setting up the nfs server")
+	prefix := oc.Namespace()
+	errs := []error{}
+
+	e2e.Logf("Adding privileged scc from system:serviceaccount:default:default")
+	if _, err := oc.AsAdmin().Run("adm").Args("policy", "add-scc-to-user", "privileged", fmt.Sprintf("system:serviceaccount:%s:default", oc.Namespace())).Output(); err != nil {
+		return nil, nil, err
+	}
+
+	e2e.Logf("Setting up the nfs server replication controller")
+	rc, err := oc.AdminKubeClient().Core().ReplicationControllers(oc.Namespace()).Create(CreateNFSServerReplicationController(fmt.Sprintf("%s%s", nfsPrefix, prefix), capacity))
+	if err != nil {
+		e2e.Logf("WARNING: unable to create replication controller %s%s: %v\n", nfsPrefix, prefix, err)
+		errs = append(errs, err)
+	}
+	err = wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
you might want to give this more than a minute, if it has to pull the image down it could take a while.  Try 5 mins.

Reply:
updated in next push